sdk/storage/Azure.Storage.Blobs/src/Generated/BlobRestClient.cs

```diff
@@ -17843,53 +17905,111 @@ public static partial class BlobsModelFactory
         /// </summary>
-        public long BlobSequenceNumber { get; internal set; }
+        public long BlobSequenceNumber { get; }
 
         /// <summary>
         /// The SHA-256 hash of the encryption key used to encrypt the pages. This header is only returned when the pages were encrypted with a customer-provided key.
         /// </summary>
-        public string EncryptionKeySha256 { get; internal set; }
+        public string EncryptionKeySha256 { get; }
 
         /// <summary>
         /// Prevent direct instantiation of PageInfo instances.
         /// You can use BlobsModelFactory.PageInfo instead.
         /// </summary>
-        internal PageInfo() { }
+        internal PageInfo(
+            Azure.Core.Http.ETag eTag,
+            System.DateTimeOffset lastModified,
+            byte[] contentHash,
+            byte[] xMSContentCrc64,
+            long blobSequenceNumber,
+            string encryptionKeySha256){
+                ETag = eTag;
+                LastModified = lastModified;
+                ContentHash = contentHash;
+                XMSContentCrc64 = xMSContentCrc64;
+                BlobSequenceNumber = blobSequenceNumber;
+                EncryptionKeySha256 = encryptionKeySha256;
+            }
+
+        /// <summary>
+        /// Check if two PageInfo instances are equal.
+        /// </summary>
+        /// <param name="other">The instance to compare to.</param>
+        /// <returns>True if they're equal, false otherwise.</returns>
+        public bool Equals(PageInfo other)
+        {
+            if (!ETag.Equals(other.ETag))
+                return false;
+            if (!LastModified.Equals(other.LastModified))
+                return false;
+            if (!ContentHash.Equals(other.ContentHash))
+                return false;
+            if (!XMSContentCrc64.Equals(other.XMSContentCrc64))
+                return false;
+            if (!BlobSequenceNumber.Equals(other.BlobSequenceNumber))
+                return false;
+            if (!EncryptionKeySha256.Equals(other.EncryptionKeySha256, System.StringComparison.InvariantCulture))
+                return false;
+
+            return true;
+        }
+
+        /// <summary>
+        /// Check if two PageInfo instances are equal.
+        /// </summary>
+        /// <param name="obj">The instance to compare to.</param>
+        /// <returns>True if they're equal, false otherwise.</returns>
+        public override bool Equals(object obj) => obj is PageInfo && Equals((PageInfo)obj);
+
+        /// <summary>
+        /// Get a hash code for the PageInfo.
+        /// </summary>
+        public override int GetHashCode(){
+            var hashCode = new Azure.Core.HashCodeBuilder();
+            hashCode.Add(ETag);
+            hashCode.Add(LastModified);
+            hashCode.Add(ContentHash);
+            hashCode.Add(XMSContentCrc64);
+            hashCode.Add(BlobSequenceNumber);
+            hashCode.Add(EncryptionKeySha256);
```

Comment:
If comparer is used in Equals the same comparer should be passed into `hashCode.Add` so the comparisons are consistent.